src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py

```diff
@@ -585,6 +463,89 @@ def closeEvent(self, event):
             # Maybe we should just minimize
             self.setWindowState(QtCore.Qt.WindowMinimized)
 
+    def on_extrapolation_text_changed_1(self, text):
+        """ Text in LowerQMax changed"""
+
+        #
+        # Note: We need to update based on params below, not a call to self.extrapolation_parameters,
+        #       because that value wont be updated until after the QLineEdit.textEdited signals are
+        #       processed
+        #
+
+        params = self.extrapolation_parmameters._replace(point_1=safe_float(text))
+        self.slider.extrapolation_parameters = params
+        self._q_space_plot.update_lines(ExtrapolationInteractionState(params))
+        self.notify_extrapolation_text_box_validity(params)
+
+    def on_extrapolation_text_changed_2(self, text):
+        """ Text in UpperQMin changed"""
+
+        #
+        # Note: We need to update based on params below, not a call to self.extrapolation_parameters,
+        #       because that value wont be updated until after the QLineEdit.textEdited signals are
+        #       processed
+        #
+
+        params = self.extrapolation_parmameters._replace(point_2=safe_float(text))
+        self.slider.extrapolation_parameters = params
+        self._q_space_plot.update_lines(ExtrapolationInteractionState(params))
+        self.notify_extrapolation_text_box_validity(params)
+
+    def on_extrapolation_text_changed_3(self, text):
+        """ Text in UpperQMax changed"""
+
+        #
+        # Note: We need to update based on params below, not a call to self.extrapolation_parameters,
+        #       because that value wont be updated until after the QLineEdit.textEdited signals are
+        #       processed
+        #
+
+        params = self.extrapolation_parmameters._replace(point_3=safe_float(text))
+        self.slider.extrapolation_parameters = params
+        self._q_space_plot.update_lines(ExtrapolationInteractionState(params))
+        self.notify_extrapolation_text_box_validity(params)
+
+    def notify_extrapolation_text_box_validity(self, params):
+        """ Set the colour of the text boxes to red if they have bad parameter definitions"""
+        box_1_style = ""
+        box_2_style = ""
+        box_3_style = ""
+        red = "QLineEdit { background-color: rgb(255,0,0); color: rgb(255,255,255) }"
+
+        if params.point_1 <= params.data_q_min:
+            box_1_style = red
+
+        if params.point_2 <= params.point_1:
+            box_1_style = red
+            box_2_style = red
+
+        if params.point_3 <= params.point_2:
+            box_2_style = red
+            box_3_style = red
+
+        # if v3 < v1 and v2, all three will go red (because of transitivity of <=), but this is good
+        if params.point_3 <= params.point_1:
```

Comment:
I wasn't sure it should be enforced. Nothing was enforced before so I had to make a judgement and went with the choice least likely to prevent people from doing what they were doing. It could be correct, I'm not sure, but I have erred on the side of letting people do what they want.